app/components/filters/include_new_or_changed.rb

```diff
@@ -0,0 +1,21 @@
+module Filters
+  class IncludeNewOrChanged
+    MAX_UPDATE_DAYS = 7 # used to limit number of entries queried to determine if an update ahs occurred.
+
+    def call
+      ETL::Filter.new(filter_name: 'Existing/Unchanged record') do |record|
+        new(record) || changed(record)
+      end
+    end
+
+    def new(record)
+      !Facts::Appointment.find_by(reference_number: record['id'])
+    end
+
+    def changed(record)
+      updated_at = Time.zone.parse(record['updated_at'])
+      updated_at > MAX_UPDATE_DAYS.days.ago &&
+        Facts::Appointment.where(reference_number: record['id']).where(['reference_updated_at < ?', updated_at]).any?
```

Comment:
We don't need to chain the `.where` calls, you can pass multiple conditions. We can also use `exists?` since we don't need to return the object.

```rb
Facts::Appointment.exists?("reference_number = ? AND reference_updated_at < ?', record['id'], updated_at])
```